src/complex/DataStructure/DataStore.hpp

```diff
@@ -49,6 +49,9 @@ class DataStore : public AbstractDataStore<T>
   DataStore(usize numTuples, std::optional<T> initValue)
   : DataStore({numTuples}, {1}, initValue)
   {
+    m_NumComponents = std::accumulate(m_ComponentShape.cbegin(), m_ComponentShape.cend(), static_cast<size_t>(1), std::multiplies<>());
+    m_NumTuples = std::accumulate(m_TupleShape.cbegin(), m_TupleShape.cend(), static_cast<size_t>(1), std::multiplies<>());
+    m_Size = m_NumTuples * m_NumComponents;
```

Comment:
This constructor already delegates to another constructor which will calculate these values.